actionpack/lib/action_controller/metal/exceptions.rb

```diff
@@ -3,14 +3,9 @@ class ActionControllerError < StandardError #:nodoc:
   end
 
   class BadRequest < ActionControllerError #:nodoc:
-    attr_reader :original_exception
-
-    def initialize(type = nil, e = nil)
-      return super() unless type && e
-
-      super("Invalid #{type} parameters: #{e.message}")
-      @original_exception = e
-      set_backtrace e.backtrace
+    def original_exception
+      ActiveSupport::Deprecation.warn("#original_exception is deprecated. Use #cause instead.")
```

Comment:
Need to pass `caller` to `#warn` so it will report the caller's file:line rather than this file's. (Ditto for other `#warn`s)